packages-exp/auth-exp/src/core/providers/oauth.ts

```diff
@@ -0,0 +1,82 @@
+/**
+ * @license
+ * Copyright 2019 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as externs from '@firebase/auth-types-exp';
+
+import { AuthErrorCode } from '../errors';
+import { debugFail } from '../util/assert';
+
+export interface CustomParameters {
+  [key: string]: string;
+}
+
+interface CredentialParameters {
+  idToken?: string;
+  accessToken?: string;
+  rawNonce?: string;
+}
+
+export class OAuthProvider implements externs.AuthProvider {
+  defaultLanguageCode: string | null = null;
+  private scopes: string[] = [];
+  private customParameters: CustomParameters = {};
+  constructor(readonly providerId: externs.ProviderId) {}
+  static credentialFromResult(
+    _userCredential: externs.UserCredential
+  ): externs.OAuthCredential | null {
+    debugFail('not implemented');
+  }
+  static credentialFromError(
+    _error: AuthErrorCode
+  ): externs.OAuthCredential | null {
+    debugFail('not implemented');
+  }
+  static credentialFromJSON(_json: object): externs.OAuthCredential {
+    debugFail('not implemented');
+  }
+
+  credential(_params: CredentialParameters): externs.OAuthCredential {
+    debugFail('not implemented');
+  }
+
+  setDefaultLanguage(languageCode: string | null): void {
+    this.defaultLanguageCode = languageCode;
```

Comment:
should this pull from navigator language like Auth does?

Reply:
Not in this method: https://github.com/firebase/firebase-js-sdk/blob/992432e07ca3b7594b860c0d50da23596311ea56/packages/auth/src/authcredential.js#L547